### PR TITLE
UCodeDisas_Disassemble

### PR DESCRIPTION
**Pull request by Random06457**
`UCodeDisas_Disassemble` is close from equivalent.

Sorry for leaving it in this state but I doubt that I'll work on it again anytime soon and at this point I feel like I'm just taking this file hostage for no reason.

### CODE REVIEW

**Review by Dragorn421 — COMMENTED** [review]
Very interesting function to try to match. It looks like maybe it could help our gbi.h towards the one that was used?

Didn't comment on formatting stuff, I think this function is special enough we can delay cleanup to whenever it's matched 🥺👉🆗

File: src/code/ucode_disas.c [hunk 1]
```diff
@@ -249,6 +249,1089 @@ void UCodeDisas_PrintVertices(UCodeDisas* this, Vtx* vtx, s32 count, s32 start)
...
 }
 
+// Todo: clean this up
+
+typedef struct {
+    s8 cmd;
+    u8 v0;
+    u8 v1;
+    u8 wd;
+} Gline3DFix;
```
[the remark this answers is not in the document]
Comment (Random06457) [reply src/code/ucode_disas.c:259]: The rest seems unused and this struct is only used only as a pointer. See: https://github.com/zeldaret/oot/blob/84324c48b6c8b4618da3d4cb6ae0059fba88fc7f/src/code/ucode_disas.c#L1018

[the remark this answers is not in the document]
Comment (Dragorn421) [reply src/code/ucode_disas.c:259]: It should probably have an explicit pad then? At least for consistency and preventing confusion

```suggestion
typedef struct {
    s8 cmd;
    u8 v0;
    u8 v1;
    u8 wd;
    u32 pad;
} Gline3DFix;
```

File: src/code/ucode_disas.c [hunk 2]
```diff
@@ -249,6 +249,1089 @@ void UCodeDisas_PrintVertices(UCodeDisas* this, Vtx* vtx, s32 count, s32 start)
...
+    u32 len : 8;
+    u32 data : 32;
+} GsetothermodeMod;
+
+typedef struct {
+    u32 cmd : 8;
+    u32 par : 8;
+    u32 len : 16;
+    u32 addr;
+} Gdma2;
```
[the remark this answers is not in the document]
Comment (Random06457) [reply src/code/ucode_disas.c:329]: I don't remember what that was for. I guess I can remove it.

File: src/code/ucode_disas.c [hunk 3]
```diff
@@ -249,6 +249,1089 @@ void UCodeDisas_PrintVertices(UCodeDisas* this, Vtx* vtx, s32 count, s32 start)
...
+    u32 f : 4;
+    u32 v : 3;
+    u32 t : 3;
+    u32 d : 3;
+    u32 y : 3;
+    u32 w : 3;
+    u32 h : 3;
+    u32 u : 3;
+    u32 s : 3;
+} GsetcombineMod;
```
[the remark this answers is not in the document]
Comment (Random06457) [reply src/code/ucode_disas.c:314]: Yes. The name come from the convention used here: https://wiki.cloudmodding.com/oot/F3DZEX2/Opcode_Details#0xFC_.E2.80.94_G_SETCOMBINE

File: src/code/ucode_disas.c [hunk 4]
```diff
@@ -249,6 +249,1089 @@ void UCodeDisas_PrintVertices(UCodeDisas* this, Vtx* vtx, s32 count, s32 start)
...
+                                    DISAS_LOG(
+                                        "/ %04x.%04x %04x.%04x %04x.%04x %.04x.%04x \\/ %12.6f %12.6f %12.6f %12.6f "
+                                        "\\\n"
+                                        "| %04x.%04x %04x.%04x %04x.%04x %.04x.%04x || %12.6f %12.6f %12.6f %12.6f |\n"
+                                        "| %04x.%04x %04x.%04x %04x.%04x %.04x.%04x || %12.6f %12.6f %12.6f %12.6f |\n"
+                                        "\\ %04x.%04x %04x.%04x %04x.%04x %.04x.%04x /\\ %12.6f %12.6f %12.6f %12.6f "
+                                        "/\n",
+                                        mtxp->intPart[0][0], mtxp->fracPart[0][0], mtxp->intPart[0][1],
+                                        mtxp->fracPart[0][1], mtxp->intPart[0][2], mtxp->fracPart[0][2],
+                                        mtxp->intPart[0][3], mtxp->fracPart[0][3], mtx.mf[0][0], mtx.mf[0][1],
```
[the remark this answers is not in the document]
Comment (Random06457) [reply src/code/ucode_disas.c:951]: That was actually wrong, not sure how I didn't catch that. Thanks!

**Review by Dragorn421 — APPROVED** [review]

File: include/functions.h [hunk 5]
```diff
@@ -1885,7 +1885,7 @@ void UCodeDisas_Destroy(UCodeDisas*);
 // ? UCodeDisas_ParseGeometryMode(?);
 // ? UCodeDisas_ParseRenderMode(?);
 // ? UCodeDisas_PrintVertices(?);
-void UCodeDisas_Disassemble(UCodeDisas*, Gfx*);
+// void UCodeDisas_Disassemble(UCodeDisas*, Gfx*);
```
Comment (fig02) [inline include/functions.h:1888]: we should remove the comments for any prototypes of functions that are not meant to be called outside of the system

Comment (Random06457) [reply include/functions.h:1888]: `UCodeDisas_Disassemble` is used in `graph.c` here
https://github.com/zeldaret/oot/blob/ceb571423db37b87336bff905078aabc6c627572/src/code/graph.c#L44
and here
https://github.com/zeldaret/oot/blob/ceb571423db37b87336bff905078aabc6c627572/src/code/graph.c#L78

Comment (fig02) [reply include/functions.h:1888]: alright, then we should remove the //

**Review by Roman971 — APPROVED** [review]
There are some things I would like to cleanup or discuss but it's not urgent either so we can probably go with this for now

**Pull request merged**